### PR TITLE
MED-50-integrate-api-for-reset-password

### PR DESCRIPTION
**Pull request by Mati-Pereira**
### 2. Handling jwt token, alowing update password only if token is valid

**PR checklist:**

**1.1 Common**
- [x] 1.1.2 no any, should be also added to eslint rules
- [x] 1.1.1 types for input and output params
- [x] 1.1.3 try/catch for any async function
- [x] 1.1.4 no magic numbers
- [x] 1.1.5 compare only with constants not with strings (instead of user === ‘admin’, better user === roles.admin)
- [x] 1.1.6 no ternary operator inside ternary operator (bad example: question ? first: second ? third: fourth)
- [x] 1.1.7 avoid similar types with duplicating by generics
- [x] 1.1.8 no console.log in pr
- [x] 1.1.9 .env file should be in .gitignore
- [x] 1.1.10 delete commented code if it's not part of documentation
- [x] 1.1.11 no links in the code, env links should be in env file (for example: server url),
- [x] constant links (for example default avatar URL) should be in constant file
- [x] 1.1.12 constants and function names should be lowercase.
- [x] 1.1.13 no dates format in the code (like "dd/MM/yyyy”), move it in global constant variable
- [x] 1.1.14 import rules. imports should come in a specific order: node modules, absolute path, relative path
- [x] 1.1.15 strings should be in the constant variable. Example: instead of ‘15 3 * * *’, const EACH_DAY_15h_15min = ‘15 3 * * *’

 **1.3 Frontend|Mobile**
- [x] 1.3.1 split component and logic (move logic to custom hook file)
- [x] 1.3.2 colors, font size, and font name should be in the theme or in the constants
- [x] 1.3.3 no text in the components, use i18n approach, even if you have only one language for now
- [x] 1.3.4 inline styles prohibited
- [x] 1.3.5 import should be absolute. instead of ../../../components/myComponent should be components/myComponent

### CODE REVIEW

File: apps/medon-fe/src/components/LoginComponent/style.tsx [hunk 1]
```diff
@@ -1,22 +1,21 @@
...
   font-size: 2rem;
   line-height: 3.5rem;
-  margin-bottom: 1rem
+  margin-bottom: 1rem;
 `;
 
 export const Text = styled.div`
-  font-family: ${theme.typography.fontFamily.sf_pro_text};
+  font-family: ${theme.fontFamily.sf_pro_text};
   font-size: 16px;
```
Comment (lyudad) [inline apps/medon-fe/src/components/LoginComponent/style.tsx]: constants

File: apps/medon-fe/src/components/LoginForm/style.tsx [hunk 2]
```diff
@@ -18,40 +17,38 @@ export const Form = styled.form`
...
-      text-align: left;
-      position: relative;
-    }
-
+  input {
+    width: 100%;
+  }
+  label {
+    font-family: ${theme.fontFamily.sf_pro_text};
+    font-size: 0.9rem;
```
Comment (lyudad) [inline apps/medon-fe/src/components/LoginForm/style.tsx]: constants. dont use rem fr fonts

File: apps/medon-fe/src/components/LoginForm/style.tsx [hunk 3]
```diff
@@ -18,40 +17,38 @@ export const Form = styled.form`
...
+  background: ${theme.colors.blue_300};
+  color: ${theme.colors.white};
+  cursor: pointer;
 `;
 
 export const StyledErrorMessage = styled.p`
   color: ${theme.colors.red_500};
-  font-family: ${(p) => p.theme.typography.fontFamily.sf_pro_text};
+  font-family: ${(p) => p.theme.fontFamily.sf_pro_text};
   font-size: 0.8rem;
```
Comment (lyudad) [inline apps/medon-fe/src/components/LoginForm/style.tsx]: same

File: apps/medon-fe/src/pages/ResetPassword/index.tsx [hunk 4]
```diff
@@ -37,42 +38,81 @@ export default function ResetPassword() {
...
+          {!isPasswordSent ? (
             <>
               <h1>{t('forget-password.reset-password.title')}</h1>
               <h3>{t('forget-password.reset-password.subtitle')}</h3>
               <Input
-                placeholder="New Password *"
+                placeholder={
+                  t(
+                    'forget-password.reset-password.placeholder-newpassword'
+                  ) as string
```
Comment (lyudad) [inline apps/medon-fe/src/pages/ResetPassword/index.tsx]: "as" is bad practice

File: apps/medon-fe/src/components/LoginComponent/style.tsx (hunk 1 above)
Comment (Mati-Pereira) [reply apps/medon-fe/src/components/LoginComponent/style.tsx]: fixed

File: apps/medon-fe/src/components/LoginForm/style.tsx (hunk 2 above)
Comment (Mati-Pereira) [reply apps/medon-fe/src/components/LoginForm/style.tsx]: fixed

File: apps/medon-fe/src/components/LoginForm/style.tsx (hunk 3 above)
Comment (Mati-Pereira) [reply apps/medon-fe/src/components/LoginForm/style.tsx]: fixed

File: apps/medon-fe/src/pages/ResetPassword/index.tsx (hunk 4 above)
Comment (Mati-Pereira) [reply apps/medon-fe/src/pages/ResetPassword/index.tsx]: fixed

**Review by lyudad — APPROVED** [review]

**Pull request merged**